src/test/regress/mitmscripts/README

```diff
@@ -0,0 +1,165 @@
+Automated Failure testing
+=========================
+
+Automated Failure Testing works by inserting a network proxy (mitmproxy) between the
+citus coordinator and one of the workers (connections to the other worker are left
+unchanged). The proxy is configurable, and sits on a fifo waiting for commands. When it
+receives a command over the fifo it reconfigures itself and sends back response.
+Regression tests which use automated failure testing communicate with mitmproxy by running
+special UDFs which talk to said fifo. The tests send commands such as "fail any connection
+which contain the string 'COMMIT'" and then run SQL queries and assert that the
+coordinator has reasonable behavior when the specified failures occur.
+
+Contents of this file:
+  I.   Getting Started
+  II.  Running mitmproxy manually
+  III. citus.mitmproxy() command strings
+  IV.  Recording Network Traffic
+
+# I. Getting Started
+
+First off, to use this you'll need mitmproxy, I recommend version 3.0.4, and I also
+recommend running it with python 3.6. This script integrates pretty deeply with mitmproxy
+so other versions might fail to work.
+
+I highly recommend using pipenv to install mitmproxy. It lets you easily manage isolated
+environments (instead of installing python packages globally). If you've heard of
+virtualenv, pipenv is that but much easier to use.
+
+Once you've installed it:
+
+$ cd src/test/regress
+$ pipenv --python 3.6
+$ pipenv install  # there's already a Pipfile.lock in src/test/regress with packages
+$ pipenv shell  # this enters the virtual environment, putting mitmproxy onto $PATH
+
+That's all you need to do to run the failure tests:
+
+$ make check-failure
+
+# II. Running mitmproxy manually
+
+$ mkfifo /tmp/mitm.fifo  # first, you need a fifo
+$ pipenv shell  # run this from inside src/test/regress
+$ mitmdump --rawtcp -p 9702 --mode reverse:localhost:9700 -s fluent.py --set fifo=/tmp/mitm.fifo
+
+The specific port numbers will be different depending on your setup. The above string
+means mitmdump will accept connections on port 9702 and forward them to the worker
+listening on port 9700.
+
+Now, open psql and run:
+
+# UPDATE pg_dist_node SET nodeport = 9702 WHERE nodeport = 9700;
+
+Again, the specific port numbers depend on your setup.
+
+# \i sql/failure_test_helpers.sql
+
+The above file creates some UDFs and also disables a few citus features which make
+connections in the background. 
+
+You also want to tell the UDFs how to talk to mitmproxy (careful, this must be an absolute
+path):
+
+# SET citus.mitmfifo = '/tmp/mitm.fifo';
```

Comment:
Shall we note that this is not actually a GUC, but, still Postgres allows us to read it? Though I'm now familiar with this, I was confused when I first saw/realize this fact.

Reply:
Sure, I can add a quick note